keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -0,0 +1,702 @@
+import os
+import warnings
+
+import numpy as np
+
+from keras.src import backend
+from keras.src import ops
+from keras.src.api_export import keras_export
+from keras.src.callbacks.monitor_callback import (
+    MonitorCallback,  # For metric monitoring logic
+)
+from keras.src.utils.io_utils import print_msg
+from keras.src.utils.module_utils import LazyModule
+
+ocp = LazyModule(
+    "orbax.checkpoint",
+    pip_name="orbax-checkpoint",
+    import_error_msg=(
+        "OrbaxCheckpoint requires the 'orbax-checkpoint' package. "
+        "Install it with: pip install orbax-checkpoint"
+    ),
+)
+
+# Note: Advanced Orbax functionality is available through the ocp LazyModule
+# Users can access it via: from keras.src.utils.module_utils import LazyModule
+# ocp = LazyModule("orbax.checkpoint"); ocp.CheckpointManager
+
+
+def _get_state_tree(model):
+    """Get the complete model state as a nested tree structure."""
+    state_tree = model.get_state_tree(value_format="numpy_array")
+
+    # Convert numpy scalar types to Python types for Orbax compatibility
+    def convert_scalars(obj):
+        if isinstance(obj, np.ndarray) and obj.ndim == 0:
+            # Convert 0-dimensional numpy arrays (scalars) to Python types
+            return obj.item()
+        elif isinstance(obj, np.generic):
+            # Convert numpy scalar types (like np.float32) to Python types
+            return obj.item()
+        elif isinstance(obj, dict):
+            return {k: convert_scalars(v) for k, v in obj.items()}
```

Comment:
In general, use `keras.tree.map_structure` to do things like that, it recurses for you.

Reply:
Done